packages/astro/vendor/vite/dist/client/client.mjs

```diff
@@ -199,6 +199,11 @@ function warnFailedFetch(err, path) {
 socket.addEventListener('message', async ({ data }) => {
     handleMessage(JSON.parse(data));
 });
+function cleanUrl(pathname) {
+  let url = new URL(pathname, location);
+  url.searchParams.delete('direct');
+  return url.pathname + url.search;
+}
```

Comment:
Would you add `/** comment */` above this to describe why it’s needed?

Reply:
https://github.com/withastro/astro/pull/2440/commits/adf3350d89c15d2676c07435c63d5829f1c72d7c

Reply:
👍